### ISSUE TITLE
"Style/MapIntoArray" false positive with push(*ary)

### ISSUE THREAD

**Issue by esotericpig** [rubocop/rubocop#13118]
I have the following code:

```ruby
a = [[1,2,3],[4,5,6]]
b = []

a.each do |x|
  b.push(*x)
end

puts a.inspect  #=> [[1, 2, 3], [4, 5, 6]]
puts b.inspect  #=> [1, 2, 3, 4, 5, 6]
```

RuboCop complains about this with `Style/MapIntoArray`, and auto-correct creates invalid code:

```
b = a.map do |x|
  *x
end
```

--------

## Expected behavior

Have `Style/MapIntoArray` enabled but not report this as an offense (or to produce correct code on auto-correct).

## Actual behavior

Reports as an offense.

## Steps to reproduce the problem

(Explained above.)

## RuboCop version

```
$ rubocop -V
1.65.1 (using Parser 3.3.4.2, rubocop-ast 1.32.0, running on ruby 3.0.2) [x86_64-linux]

$ ruby -v
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux]
```

**Comment by Earlopain** [comment]
Thanks for the report! This will be fixed in the next version by not registering an offense when the argument is getting splatted.

**Comment by esotericpig** [comment]
Great, thanks very much!
